explainaboard/utils/span_utils.py

```diff
@@ -156,318 +225,113 @@ def get_matched_spans(
-                    span.span_efre = self.resources["efre_dic"].get(lower_text, 0.0)
-
-                spans.append(span)
-                span_type, span_start = None, -1
-
-            # End of a span + start of a span!
-            elif tok != default:
-
-                tok_span_class, tok_span_type = tok.split('-')
-                if span_type is None:
-                    span_type, span_start = tok_span_type, i
-                elif tok_span_type != span_type or tok_span_class == "B":
-                    span_text = " ".join(seq[span_start:i])
-                    span = Span(
-                        span_text=span_text,
-                        span_tag=span_type,
-                        span_pos=(span_start, i),
-                        span_capitalness=cap_feature(span_text),
-                        span_rel_pos=i * 1.0 / len(seq),
-                        span_chars=len(span_text),
-                        span_tokens=len(span_text.split(" ")),
-                    )
-                    if (
-                        "has_stats" in self.resources.keys()
-                        and self.resources["has_stats"]
-                    ):
-                        lower_tag = span.span_tag.lower()
-                        lower_text = span.span_text.lower()
-                        span.span_econ = self.resources["econ_dic"].get(
-                            f'{lower_text}|||{lower_tag}', 0.0
-                        )
-                        span.span_efre = self.resources["efre_dic"].get(lower_text, 0.0)
-                    spans.append(span)
-                    span_type, span_start = tok_span_type, i
-            else:
-                pass
-        # end condition
-        if span_type is not None:
-            span_text = " ".join(seq[span_start : len(tags)])
-            span = Span(
-                span_text=span_text,
-                span_tag=span_type,
-                span_pos=(span_start, len(tags)),
-                span_capitalness=cap_feature(span_text),
-                span_rel_pos=len(tags) * 1.0 / len(seq),
-                span_chars=len(span_text),
-                span_tokens=len(span_text.split(" ")),
-            )
-
-            if "has_stats" in self.resources.keys() and self.resources["has_stats"]:
-                lower_tag = (
-                    span.span_tag.lower()
-                    if span.span_tag is not None
-                    else span.span_tag
-                )
-                lower_text = (
-                    span.span_text.lower()
-                    if span.span_text is not None
-                    else span.span_text
-                )
-                span.span_econ = self.resources["econ_dic"].get(
-                    f'{lower_text}|||{lower_tag}', 0.0
-                )
-                span.span_efre = self.resources["efre_dic"].get(lower_text, 0.0)
-            spans.append(span)
-
-        return spans
+    def _span_ends(self, tags: list[str], i: int) -> bool:
```

Comment:
the introduced strategy of using `_span_ends` and `_span_start` is surprisingly fantastic

Reply:
Learned: `if one does not find a unified method, it is just because the disassembly is not detailed enough`